src/Cache/CacheOperationTypes/CacheOperationTypes.php

```diff
@@ -441,3 +475,209 @@ class CacheListPushBackResponseError extends CacheListPushBackResponse
     use ErrorBody;
 }
 
+abstract class CacheListPopFrontResponse extends ResponseBase
+{
+    public function asHit() : CacheListPopFrontResponseHit|null
+    {
+        if ($this->isHit())
+        {
+            return $this;
+        }
+        return null;
+    }
+
+    public function asMiss() : CacheListPopFrontResponseMiss|null
+    {
+        if ($this->isMiss())
+        {
+            return $this;
+        }
+        return null;
+    }
+
+    public function asError() : CacheListPopFrontResponseError|null
+    {
+        if ($this->isError())
+        {
+            return $this;
+        }
+        return null;
+    }
+}
+
+class CacheListPopFrontResponseHit extends CacheListPopFrontResponse {
+    private string $value;
+
+    public function __construct(_ListPopFrontResponse $response)
+    {
+        parent::__construct();
+        $this->value = $response->getFound()->getFront();
+    }
+
+    public function value() : string
+    {
+        return $this->value;
+    }
```

Comment:
Are we only doing `string` values here as opposed to both `string` and `bytes`?

Reply:
Yes, for reasons we discussed today.